src/cargo/ops/tree/graph.rs

```diff
@@ -351,6 +351,13 @@ fn add_pkg(
                 true
             })
             .collect();
+
+        // This dependency is eliminated from the dependency tree under
+        // the current target and feature set.
+        if deps.is_empty() {
```

Comment:
Do you know what triggered this? I'm curious why this didn't show up  before and only now is it being added.

Reply:
This was added because otherwise the line
```rust
let dep_pkg = graph.package_map[&dep_id]
```
slightly further down would panic. The reason for this working in the past is that we would always have the `.crate` files for all dependencies, and so `package_map` would have an entry for every dep, even ones that are not actually enabled by the current build. With this fix however, we _don't_ populate `package_map` with dependencies that aren't activated, and so we have to make sure we don't do the lookup if there are no activation paths (i.e., if `deps.is_empty()`).